common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -378,13 +378,6 @@ public boolean matchAt(final UTF8String s, int pos) {
     return ByteArrayMethods.arrayEquals(base, offset + pos, s.base, s.offset, s.numBytes);
   }
 
-  private boolean matchAt(final UTF8String s, int pos, int collationId) {
-    if (s.numBytes + pos > numBytes || pos < 0) {
-      return false;
-    }
-    return this.substring(pos, pos + s.numBytes).semanticCompare(s, collationId) == 0;
```

Comment:
I wonder can't we just use `CollationFactory.getStringSearch` here?

Reply:
We can, that's exactly what Stevo is doing - removing this code and replacing it with `CollationFactory.getStringSearch`. For more context: the old implementation was there before we introduced `StringSearch`.

Reply:
Migrated `collatedStartsWith` and `collatedEndsWith` functions to here